cloudflare.go

```diff
@@ -464,13 +481,41 @@ func (worker *CloudflareWorker) UpdateIPLists() error {
 		}
 	}
 
-	for action, state := range newIPSetByAction {
-		if reflect.DeepEqual(worker.CFStateByAction[action].IPListState.IPSet, state) || len(state) == 0 {
+	for action, set := range newIPSetByAction {
+		if reflect.DeepEqual(worker.CFStateByAction[action].IPListState.IPSet, set) {
 			log.Info("no changes to IP rules ")
 			continue
 		}
+		if len(set) == 0 {
+			// The ReplaceIPListItemsAsync method doesn't allow to empty the list.
+			// Hence we only add one mock IP and later delete it.
+			set["10.0.0.1"] = struct{}{}
+			defer func(action string) {
+				ipListId := worker.CFStateByAction[action].IPListState.IPList.ID
+				items, err := worker.getAPI().ListIPListItems(worker.Ctx, ipListId)
+				if err != nil {
+					worker.Logger.Error(err)
+					return
+				}
+				_, err = worker.getAPI().DeleteIPListItems(worker.Ctx, ipListId, cloudflare.IPListItemDeleteRequest{
+					Items: []cloudflare.IPListItemDeleteItemRequest{
+						{
+							ID: items[0].ID,
+						},
+					},
+				})
+				if err != nil {
+					worker.Logger.Error(err)
+				}
+				worker.CFStateByAction[action].IPListState.IPSet = make(map[string]struct{})
+				worker.CFStateByAction[action].IPListState.IPList.NumItems = 0
+				worker.UpdatedState <- worker.CFStateByAction
+				worker.Logger.Info("removed mock IP")
```

Comment:
maybe change message to "emptied ip list"

Reply:
Fixed via https://github.com/crowdsecurity/cs-cloudflare-bouncer/pull/56/commits/8b56fccf8f992071713099b8d387df061530fcee